hs/Database/CourseQueries.hs

```diff
@@ -124,6 +125,16 @@ courseInfo = do
             Just $ Tables.Session (map buildLecture lectures)
                                   (map buildTutorial tutorials)
 
+-- | Return a list of all departments
+deptList :: IO Response
+deptList = do
+    depts <- runSqlite dbStr $ do
+        courses :: [Entity Courses] <- selectList [] []
+        return $ sort . nub $ map f courses
+    return $ createJSONResponse $ encodeJSON $ Aeson.toJSON depts
```

Comment:
We seem to use `encodeJSON` along with `createJSONResponse` in both uses of `createJSON` response. Should we absorb encodeJSON in createJSONResponse? We also use Aeson.toJSON in both as well.